### PR TITLE
Const div implementation

### PR DESCRIPTION
**Pull request by kaidokert**
Div part of c0nstification #58

## Summary by Sourcery

Implement const-evaluable division for FixedUInt and update internal helpers to use array-based const operations.

New Features:
- Add const-compatible division routine operating on FixedUInt backing arrays.

Enhancements:
- Refactor FixedUInt division implementation to delegate to shared const_div logic.
- Introduce const_sub_shifted helper for subtracting shifted divisors in-place on word arrays.
- Adapt helper-method tests to exercise const-based bit, compare, and subtract operations on underlying arrays.

Tests:
- Update and extend helper method tests to validate const_cmp_shifted, const_sub_shifted, and const_set_bit behavior across edge cases.

## Summary by CodeRabbit

* **Breaking Changes**
  * Removed several utility methods from fixed-size unsigned integer handling.

* **Improvements**
  * Added broader compile-time (const) evaluation support for arithmetic operations.
  * Reworked division to use a const-friendly path that yields quotient in-place and returns remainder.
  * Introduced const helpers for array-based arithmetic to enable const-context algorithms.

* **Tests**
  * Updated test suite and scaffolding to use the new const-based helpers.

### CODE REVIEW

File: src/fixeduint.rs [hunk 1]
```diff
@@ -350,117 +350,12 @@ c0nst::c0nst! {
...
-            if dividend.cmp_shifted(divisor, bit_pos) != core::cmp::Ordering::Less {
-                dividend.sub_shifted(divisor, bit_pos);
-                quotient.set_bit(bit_pos);
-            }
-
-            if bit_pos == 0 {
-                break;
-            }
-            bit_pos -= 1;
+        let remainder_array = const_div(&mut dividend.array, &divisor.array);
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint.rs]: There is no shadowing, this isn't in an `impl` block, it's in `c0nst` block

File: src/fixeduint.rs [hunk 2]
```diff
@@ -826,6 +721,113 @@ c0nst::c0nst! {
...
+        // dividend == divisor: quotient = 1, remainder = 0
+        if matches!(const_cmp::<T, N>(dividend, divisor), Ordering::Equal) {
+            let mut i = 0;
+            while i < N {
+                dividend[i] = <T as ConstZero>::zero();
+                i += 1;
+            }
+            if N > 0 {
+                dividend[0] = <T as ConstOne>::one();
+            }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint.rs]: micro opt in const code. not really
